### ISSUE TITLE
Gradient computation of rotation_matrix_to_angle_axis may have something wrong

### ISSUE THREAD

**Issue by sh8** [kornia/kornia#1267]
### Describe the bug

I was trying to perform gradient optimization using SMPLify through `rotation_matrix_to_angle_axis`. However, it stopped working after I started using `rotation_matrix_to_angle_axis` in Kornia instead of one in Pytorch Geometry. After careful investigation, I found the gradient of the function seems wrong and it resulted in a failure of converging SMPL body and pose parameters. My implementation which bypasses `rotation_matrix_to_quaternion` and `quaternion_to_angle_axis` works well without any problems, so there might be a potential bug in these to quaternion-related functions.

### Reproduction steps

```bash
Please run the following code. `rm2aa` is a fixed version of the `rotation_matrix_to_angle_axis ` function.


import torch
from kornia.geometry.conversions import (
    rotation_matrix_to_angle_axis as rm2aa,
    angle_axis_to_rotation_matrix as aa2rm
)


def rm2aa_fixed(matrix: torch.Tensor, eps: float = 1.0e-8):
    bs = matrix.shape[0]

    # Axes.
    axis = torch.zeros((bs, 3), device=matrix.device)

    axis[:, 0] = matrix[:, 2, 1] - matrix[:, 1, 2]
    axis[:, 1] = matrix[:, 0, 2] - matrix[:, 2, 0]
    axis[:, 2] = matrix[:, 1, 0] - matrix[:, 0, 1]

    # Angle.
    norm = torch.norm(axis[:, 1:], dim=1).unsqueeze(1)
    r = torch.cat((axis[:, 0].unsqueeze(1), norm), dim=1)
    r = torch.norm(r, dim=1).unsqueeze(1) + eps
    t = matrix[:, 0, 0] + matrix[:, 1, 1] + matrix[:, 2, 2]
    t = t.unsqueeze(1)
    theta = torch.atan2(r, t - 1)

    # Normalise the axis.
    axis = axis / r

    # Return the data.
    return theta * axis


if __name__ == '__main__':
    R = torch.tensor([[[0.9734, -0.2059, 0.1006], [0.1948, 0.9747, 0.1099],
                       [-0.1207, -0.0874, 0.9888]]], requires_grad=True)

    print('This test is to validate the fixed-version implementation and cycle-consistency\n')
    aa1 = rm2aa_fixed(R)
    aa2 = rm2aa(R)
    R1 = aa2rm(aa1)
    R2 = aa2rm(aa2)
    print('Angle Axis Fixed - Angle Axis Orig')
    print(aa1 - aa2)
    print('Rotation Matrix Fixed - Rotation Matrix Orig')
    print(R1 - R2)
    print('Everyting looks good so far!!\n')

    print('Check if the leaf gradient is correct...')
    aa1.sum().backward()
    grad1 = R.grad.clone()

    R.grad.data.zero_()

    aa2.sum().backward()
    grad2 = R.grad.clone()

    print(grad1 - grad2)
    print('Gradient is different...')
```
```


### Expected behavior

Leaf gradient should be the same

### Environment

```shell
wget https://raw.githubusercontent.com/pytorch/pytorch/master/torch/utils/collect_env.py
# For security purposes, please check the contents of collect_env.py before running it.
python collect_env.py
```
- PyTorch Version (e.g., 1.0): 1.9
- OS (e.g., Linux): Arch Linux
- How you installed PyTorch (`conda`, `pip`, source): pip
- Build command you used (if compiling from source): 
- Python version: 3.8.5
- CUDA/cuDNN version: 10.1
- GPU models and configuration: RTX2080 Super
- Any other relevant information: None
```


### Additional context

_No response_

**Comment by edgarriba** [comment]
@sh8 thanks for reporting that ! are you open to upgrade the current function with yours in a PR ? /cc @amonszpart did some fixes early this year on that

**Comment by sh8** [comment]
thank you for the quick reply! yes, sure!

**Comment by sh8** [comment]
I made a PR! Could you please check it when you have time?
https://github.com/kornia/kornia/pull/1270

**Comment by edgarriba** [comment]
If anyone is interested in this, feel free to re-open the ticket. Thanks!
